BraintreePayPal/PayPalDataCollector/PPDataCollector.m

```diff
@@ -62,4 +69,21 @@ + (nonnull NSString *)collectPayPalDeviceData {
     return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
 }
 
+
++ (nonnull NSString *)collectPayPalDeviceInfoWithClientMetadataID:(nullable NSString *)clientMetadataID
+{
+    NSDictionary *payloadDict = [[PPDataCollector generateMagnesResultWithClientMetadataID:clientMetadataID disableBeacon:NO data:nil] getDeviceInfo];
+    if (payloadDict == nil) {
+        return @"";
+    }
+    NSError *error;
+
+    NSData *data = [NSJSONSerialization dataWithJSONObject:payloadDict options:0 error:&error];
+    if (!data) {
+        NSLog(@"ERROR: Failed to get PayPal DeviceInfo , error = %@", error);
+    }
+    return [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
```

Comment:
NSString's `initWithData:` method requires a non-null `data` value. This should probably return early if `NSJSONSerialization` fails and `data` is null, similar to the early return when `payloadDict` is null.